### PR TITLE
[wptrunner] Add `id_hash` chunk type

### PR DESCRIPTION
**Pull request by jonathan-j-lee**
Currently, `wpt run --chunk-type=hash` assigns tests to chunks using a hash of the test path. This scheme can sometimes unevenly distribute work when a slow test generates many variants or `.any.js` scopes, which are always bucketed into the same chunk. For CI systems that use chunking, uneven partitioning can contribute to greater long-pole times and longer overall builds.

This change introduces a new `id_hash` strategy that hashes by test ID for maximum uniformity.

### CODE REVIEW

**Comment by jonathan-j-lee** [comment]
@jgraham Any reason not to do this? [This change](https://chromium-review.googlesource.com/c/chromium/src/+/4434792?tab=checks) appears to reduce max shard runtimes in Chromium CI.

Also, I didn't think this met the threshold for an RFC, but let me know if you think otherwise.

**Review by WeizhongX — APPROVED** [review]

File: infrastructure/metadata/infrastructure/testdriver/file_upload.sub.html.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [file_upload.sub.html]
   expected:
-    if product == "safari": TIMEOUT
+    if product == "safari": [OK, TIMEOUT]
```
Comment (jgraham) [inline infrastructure/metadata/infrastructure/testdriver/file_upload.sub.html.ini:3]: Is this change actually required?

**Review by jgraham — COMMENTED** [review]
Looks OK to me, but it would be good to avoid the unrelated metadata change if possible (if it's needed let's land it in a separate PR).

File: infrastructure/metadata/infrastructure/testdriver/file_upload.sub.html.ini (hunk 1 above)
Comment (jonathan-j-lee) [reply infrastructure/metadata/infrastructure/testdriver/file_upload.sub.html.ini]: Maybe not, so I won't pursue this for now. When I was drafting this change, the macOS `infrasturcture/` check for STP reported `UNEXPECTED-OK` for this test on multiple pushes, but my latest push didn't reproduce that failure.

Comment (jonathan-j-lee) [reply infrastructure/metadata/infrastructure/testdriver/file_upload.sub.html.ini]: I've reencountered `UNEXPECTED-OK` in other changes, so I landed #40095.
